packages/blade/src/components/Avatar/_decisions/decisions.md

```diff
@@ -0,0 +1,170 @@
+# Avatar
+
+An avatar component is a standardized visual representation of a user or entity. This reusable element, often manifesting as a profile picture, icon, or initials, facilitates user recognition and streamlines interface navigation.
+
+This document serves as an overview of the API for the `Avatar` component.
+
+
+<img width="50%" alt="Avatar Thumbnail" src="https://github.com/razorpay/blade/assets/46647141/6e89d932-602f-435f-9519-c46a41ddb4d5" />
+
+<img width="50%" alt="AvatarGroup Thumbnail" src="https://github.com/razorpay/blade/assets/46647141/e54260e8-7f8c-4343-b5f9-8b332d65161f" />
+
+
+## Design
+
+- [Explore the design in Figma: Avatar](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=88229-1518352&mode=design&t=Gp3eolSGw8SybZkM-11)
+
+## API
+
+The `Avatar` & `AvatarGroup` components would have the following props:
+
+```ts
+type AvatarGroupProps = {
+  /**
+   * Children elements representing the avatars to stack.
+   */
+  children: React.ReactNode;
+  /**
+   * The size of each avatar within the group. Propagates to all avatars.
+   */
+  size?: 'xsmall' | 'small' | 'medium' | 'large';
+  /**
+   * The maximum number of avatars to display before truncating.
+   */
+  maxAvatars?: number;
+};
+
+type AvatarProps = {
+  /**
+   * The content or children of the avatar.
+   */
+  children?: StringChildrenType;
+  /**
+   * The size of the avatar.
+   */
+  size?: 'xsmall' | 'small' | 'medium' | 'large';
+  /**
+   * The visual variant of the avatar.
+   */
+  variant?: 'circle' | 'square';
+  /**
+   * The color theme of the avatar.
+   */
+  color?: 'primary' | 'positive' | 'negative' | 'notice' | 'information' | 'neutral';
+  /**
+   * Custom icon component to use as the avatar.
+   */
+  icon?: IconComponent;
+  /**
+   * The name of the avatar, used to generate initials.
+   * If src has loaded, the name will be used as the alt attribute of the img. If src is not loaded, the name will be used to create the initials.
+   */
+  name?: string;
+  /**
+   * Custom image source for an image avatar.
+   */
+  src?: string;
```

Comment:
What is the order of precedence among these three if the consumer provides all of them?

Reply:
`src > icon > name` will be the priority.